Source/Core/UICommon/DiscordPresence.cpp

```diff
@@ -244,6 +246,21 @@ void UpdateDiscordPresence(int party_size, SecretType type, const std::string& s
       // Note: joining still works without partyMax
     }
   }
+#ifdef USE_RETRO_ACHIEVEMENTS
+  else if (Config::Get(Config::RA_ENABLED) && Config::Get(Config::RA_RICH_PRESENCE_ENABLED) &&
+           Config::Get(Config::RA_DISCORD_PRESENCE_ENABLED))
+  {
+    AchievementManager::RichPresence rich_presence =
+        AchievementManager::GetInstance().GetRichPresence();
+    std::string state_string(rich_presence.data());
+    if (state_string.length() >= 128)
+    {
+      // 124 characters + 3 dots + null terminator - thanks to Stenzek for format
+      state_string.replace(124, 4, "...");
+    }
+    discord_presence.state = state_string.c_str();
```

Comment:
This will result in a stale pointer being assigned to the `state` member, since `state_string` will no longer exist after this scope is exited.